cocos/network/SocketIO.cpp

```diff
@@ -57,32 +52,32 @@ class SocketIOPacketV10x;
 class SocketIOPacket
 {
 public:
-    typedef enum
+    enum class SocketIOVersion
     {
         V09x,
         V10x
-    }SocketIOVersion;
+    };
 
     SocketIOPacket();
     virtual ~SocketIOPacket();
-    void initWithType(std::string packetType);
+    void initWithType(const std::string& packetType);
     void initWithTypeIndex(int index);
 
-    std::string toString();
-    virtual int typeAsNumber();
-    std::string typeForIndex(int index);
+    std::string toString()const;
+    virtual int typeAsNumber()const;
+    const std::string& typeForIndex(int index)const;
 
-    void setEndpoint(std::string endpoint){ _endpoint = endpoint; };
-    std::string getEndpoint(){ return _endpoint; };
+    void setEndpoint(const std::string& endpoint){ _endpoint = endpoint; };
+    const std::string& getEndpoint()const{ return _endpoint; };
     void setEvent(std::string event){ _name = event; };
-    std::string getEvent(){ return _name; };
+    const std::string& getEvent()const{ return _name; };
 
-    void addData(std::string data);
-    std::vector<std::string> getData(){ return _args; };
-    virtual std::string stringify();
+    void addData(const std::string& data);
+    std::vector<std::string> getData()const{ return _args; };
+    virtual std::string stringify()const;
 
-    static SocketIOPacket * createPacketWithType(std::string type, SocketIOPacket::SocketIOVersion version);
-    static SocketIOPacket * createPacketWithTypeIndex(int type, SocketIOPacket::SocketIOVersion version);
+    static SocketIOPacket * createPacketWithType(std::string type, SocketIOVersion version);
```

Comment:
We can't add `const` here since the function `createPacketWithType` will modify the state of the object.